zerver/tests/test_subs.py

```diff
@@ -1974,7 +1975,7 @@ def test_multi_user_subscription(self) -> None:
                     streams_to_sub,
                     dict(principals=ujson.dumps([user1.email, user2.email])),
                 )
-        self.assert_length(queries, 43)
+        self.assert_length(queries, 44)
```

Comment:
I'm not sure about this change so it needs to be reviewed.

Reply:
This change is definitely implied by your implementation of `public_stream_user_ids`, which does a database query.  But I'm not sure that change is quite right.

Reply:
Yeah, I think this will go away if you make the change I suggest above.